### PR TITLE
begin integrating localup script with test-cmd

### PR DESCRIPTION
**Pull request by juanvallejo**
~~Work in progress... Opening now to collect feedback~~

~~Depends on https://github.com/openshift/origin/pull/21221~~

## Issues to address as a result of this Pull

- [ ] https://bugzilla.redhat.com/show_bug.cgi?id=1639449 `image-lookup.sh` tests need to be re-enabled
- [ ] https://bugzilla.redhat.com/show_bug.cgi?id=1639449 `image.sh:71` 
- [ ] https://bugzilla.redhat.com/show_bug.cgi?id=1639453 `explain.sh` tests need to be fixed to work with openshift types. They need to then be re-enabled
- [ ] https://bugzilla.redhat.com/show_bug.cgi?id=1639461 `observe.sh:71-72` tests need to be reverted to use the original `172.30.0.0/16` CIDR
- [ ] https://github.com/openshift/origin/issues/21259 `etcdhelper` tool needs to be removed from the Origin repo. test-cmd tests for this tool were removed in this pull.
- [ ] https://bugzilla.redhat.com/show_bug.cgi?id=1639835 `SDN tests no longer able to run without network APIs`

cc @deads2k @mfojtik

### CODE REVIEW

**Comment by juanvallejo** [comment]
Apparently the /healthz/ready endpoint never becomes available.
I am able to do `oc get --raw /healthz --as system:unauthenticated`. Hitting the `/healthz/ready` endpoint gives the error:

```
Error from server (NotFound): the server could not find the requested resource
```

**Comment by juanvallejo** [comment]
cc @tnozicka

**Comment by juanvallejo** [comment]
Now able to create registry resources when running test-cmd locally.
Need to get cmd tests to actually show up as failing in CI. 
@stevekuznetsov I'm not sure how to get `test-cmd.sh` to exit with code 1 whenever there's a failure in the sourced `./hack/local-up-master/lib.sh` file.

**Comment by deads2k** [comment]
@juanvallejo given that it appears frozen on a delete, the namespace lifecycle controller in kube-controller-manager is a log that I would look at

**Comment by juanvallejo** [comment]
the /version/openshift appears to be missing

**Comment by juanvallejo** [comment]
test-cmd errors remaining: https://gist.github.com/juanvallejo/e4268b57aaac8b2c438ad20be38ed02a

File: test/cmd/dns.sh [hunk 1]
```diff
@@ -1,48 +0,0 @@
-#!/bin/bash
```
[the remark this answers is not in the document]
Comment (deads2k) [reply test/cmd/dns.sh]: > cc @ironcladlou for approval on this deletion

More like notification.  The kube-apiserver doesn't include a DNS server.

File: test/cmd/image-lookup.sh [hunk 2]
```diff
@@ -15,6 +15,9 @@ project="$( oc project -q )"
 os::test::junit::declare_suite_start "cmd/image-lookup"
 # This test validates image lookup resolution
 
+# TODO: fix and re-enable these tests before 4.0 release
+exit 0
```
Comment (juanvallejo) [inline test/cmd/image-lookup.sh]: @deads2k Created https://bugzilla.redhat.com/show_bug.cgi?id=1639449 to track the fix for these tests

File: test/cmd/explain.sh [hunk 3]
```diff
@@ -10,6 +10,8 @@ trap os::test::junit::reconcile_output EXIT
 ) &>/dev/null
 
 
+exit 0
```
Comment (juanvallejo) [inline test/cmd/explain.sh]: @deads2k Created https://bugzilla.redhat.com/show_bug.cgi?id=1639453 to track the fix for these tests

File: test/cmd/images.sh [hunk 4]
```diff
@@ -67,7 +67,8 @@ os::cmd::expect_success_and_text 'oc get istag/test:new -o jsonpath={.image.dock
 
 repository="$( oc get is/test -o jsonpath='{.status.dockerImageRepository}' )"
 os::cmd::expect_success "oc annotate --context='${cluster_admin_context}' --overwrite image/sha256:b2f400f4a5e003b0543decf61a0a010939f3fba07bafa226f11ed7b5f1e81237 images.openshift.io/deny-execution=true"
-os::cmd::expect_failure_and_text "oc run vulnerable --image=${repository}:new --restart=Never" 'spec.containers\[0\].image: Forbidden: this image is prohibited by policy'
+# TODO: re-enable before 4.0 release
```
Comment (juanvallejo) [inline test/cmd/images.sh]: This is being tracked in the same bugzilla as image-lookup tests: https://bugzilla.redhat.com/show_bug.cgi?id=1639449#c1

File: test/cmd/observe.sh [hunk 5]
```diff
@@ -28,8 +28,8 @@ os::cmd::expect_failure_and_text 'oc observe services --exit-after=1m --all-name
 os::cmd::expect_failure_and_text 'oc observe services --exit-after=1m --all-namespaces --retry-on-exit-code=2 --maximum-errors=1 --loglevel=4 -- /bin/sh -c "exit 2"' 'retrying command: exit status 2'
 
 # argument templates
-os::cmd::expect_success_and_text 'oc observe services --once --all-namespaces -a "{ .spec.clusterIP }"' '172.30.0.1'
-os::cmd::expect_success_and_text 'oc observe services --once --all-namespaces -a "{{ .spec.clusterIP }}" --output=gotemplate' '172.30.0.1'
+os::cmd::expect_success_and_text 'oc observe services --once --all-namespaces -a "{ .spec.clusterIP }"' '10.3.0.1'
```
Comment (juanvallejo) [inline test/cmd/observe.sh]: @deads2k Tracking these changes in https://bugzilla.redhat.com/show_bug.cgi?id=1639461

File: test/cmd/dns.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (ironcladlou) [reply test/cmd/dns.sh]: These tests look like a less comprehensive version of the origin DNS e2e tests which are themselves now probably superseded by upstream e2e tests, so I'm not sure there's any good reason to keep these. /cc @smarterclayton since he seems to know the most about the existing DNS tests.

File: test/cmd/status.sh [hunk 6]
```diff
@@ -21,38 +21,38 @@ cp "${KUBECONFIG}" "${login_kubeconfig}"
...
 
 # make sure `oc status` re-uses the correct "no projects" message from `oc login` with no self-provisioner role
 os::cmd::expect_success_and_text 'oc status' "You don't have any projects. Contact your system administrator to request a project"
 os::cmd::expect_success_and_text 'oc status --all-namespaces' "Showing all projects on server"
 # make sure standard login prompt is printed once self-provisioner status is restored
 os::cmd::expect_success "oc logout"
 os::cmd::expect_success "oc adm policy add-cluster-role-to-group self-provisioner system:authenticated:oauth --kubeconfig='${login_kubeconfig}'"
-os::cmd::expect_success_and_text "oc login --server=${KUBERNETES_MASTER} --certificate-authority='${MASTER_CONFIG_DIR}/ca.crt' -u test-user -p anything" "You don't have any projects. You can try to create a new project, by running"
+os::cmd::expect_success_and_text "oc login --server=${KUBERNETES_MASTER} --certificate-authority='${MASTER_CONFIG_DIR}/server-ca.crt' -u test-user2 -p anything" "You don't have any projects. You can try to create a new project, by running"
```
Comment (juanvallejo) [inline test/cmd/status.sh]: @deads2k this test is failing here now. For some reason, `test-user2` is not considered part of the `system:authenticated:oauth` group after logging in here. I have verified that `test-user2` does have a token at this point.

File: hack/test-cmd.sh [hunk 7]
```diff
@@ -54,28 +56,41 @@ os::log::system::start
...
 
-os::start::configure_server
+eval $( ./hack/install-etcd.sh | tail -n1 )
+localup::init_master
 
 os::test::junit::declare_suite_start "cmd/version"
-os::cmd::expect_success_and_not_text "KUBECONFIG='${MASTER_CONFIG_DIR}/admin.kubeconfig' oc version" "did you specify the right host or port"
-os::cmd::expect_success_and_not_text "KUBECONFIG='' oc version" "Missing or incomplete configuration info"
+os::cmd::expect_success_and_not_text "KUBECONFIG='${CONFIG}/admin.kubeconfig' oc version" "did you specify the right host or port"
+#os::cmd::expect_success_and_not_text "KUBECONFIG='' oc version" "Missing or incomplete configuration info"
```
Comment (soltysh) [inline hack/test-cmd.sh]: Why this is commented out?

File: hack/local-up-master/lib.sh [hunk 8]
```diff
@@ -64,9 +61,14 @@ clusterup::cleanup() {
   ${USE_SUDO:+sudo} fuser -k 2379/tcp
 }
 
+function clusterup::cleanup_config() {
+    echo "Deleting masterup config directory at ${LOCALUP_CONFIG}..."
```
Comment (soltysh) [inline hack/local-up-master/lib.sh]: Debugging leftover.

File: hack/local-up-master/openshift-apiserver-manifests/03_apiservice.yaml [hunk 9]
```diff
@@ -53,19 +53,6 @@ items:
     insecureSkipTLSVerify: true
     groupPriorityMinimum: 20000
     versionPriority: 15
-- apiVersion: apiregistration.k8s.io/v1
```
Comment (soltysh) [inline hack/local-up-master/openshift-apiserver-manifests/03_apiservice.yaml]: Any particular reason you're removing it?

File: hack/test-cmd.sh [hunk 10]
```diff
@@ -8,10 +8,12 @@ source "$(dirname "${BASH_SOURCE}")/local-up-master/lib.sh"
 
 function cleanup() {
   return_code=$?
-  clusterup::cleanup
+  echo "test-cmd exiting with code $return_code"
```
Comment (soltysh) [inline hack/test-cmd.sh]: Debugging leftover.

File: hack/test-cmd.sh [hunk 11]
```diff
@@ -127,6 +134,7 @@ os::log::debug "Metrics information logged to ${LOG_DIR}/metrics.log"
 oc get --raw /metrics --kubeconfig="${MASTER_CONFIG_DIR}/admin.kubeconfig"> "${LOG_DIR}/metrics.log"
 
 if [[ -n "${failed:-}" ]]; then
+    echo "test-cmd: not ok"
```
Comment (soltysh) [inline hack/test-cmd.sh]: Debugging leftover.

File: test/cmd/admin.sh [hunk 12]
```diff
@@ -14,8 +14,8 @@ trap os::test::junit::reconcile_output EXIT
   oc delete groups/orphaned-group
   oc delete users/cascaded-user
   oc delete users/orphaned-user
-  oc delete identities/anypassword:orphaned-user
-  oc delete identities/anypassword:cascaded-user
+  oc delete identities/alwaysallow:orphaned-user
+  oc delete identities/alwaysallow:cascaded-user
```
Comment (soltysh) [inline test/cmd/admin.sh]: Why changing this?

File: test/cmd/admin.sh [hunk 13]
```diff
@@ -24,18 +24,19 @@ project="$( oc project -q )"
 defaultimage="openshift/origin-\${component}:latest"
 USE_IMAGES=${USE_IMAGES:-$defaultimage}
 
-export NODECONFIG="${NODE_CONFIG_DIR}/node-config.yaml"
+#export NODECONFIG="${NODE_CONFIG_DIR}/node-config.yaml"
```
Comment (soltysh) [inline test/cmd/admin.sh]: Just remove.

File: test/cmd/admin.sh [hunk 14]
```diff
@@ -416,8 +417,8 @@ os::cmd::expect_success 'oc adm prune auth user/cascaded-user'
 os::cmd::expect_success 'oc delete user  cascaded-user'
 os::cmd::expect_success 'oc delete user  orphaned-user  --cascade=false'
 # Verify all identities remain
-os::cmd::expect_success 'oc get identities/anypassword:cascaded-user'
-os::cmd::expect_success 'oc get identities/anypassword:orphaned-user'
+os::cmd::expect_success 'oc get identities/alwaysallow:cascaded-user'
+os::cmd::expect_success 'oc get identities/alwaysallow:orphaned-user'
```
Comment (soltysh) [inline test/cmd/admin.sh]: Why this change?

File: test/cmd/admin.sh [hunk 15]
```diff
@@ -24,18 +24,19 @@ project="$( oc project -q )"
...
 # This test validates admin level commands including system policy
 
 os::test::junit::declare_suite_start "cmd/admin/start"
 # Check failure modes of various system commands
 os::cmd::expect_failure_and_text 'openshift-sdn' 'you must specify a configuration file with --config'
-os::cmd::expect_failure_and_text 'openshift-sdn --config=${NODECONFIG} --enable=kubelet' 'the following components are not recognized: kubelet'
-os::cmd::expect_failure_and_text 'openshift-sdn --config=${NODECONFIG} --enable=kubelet,other' 'the following components are not recognized: kubelet, other'
-os::cmd::expect_failure_and_text 'openshift-sdn --config=${NODECONFIG} --disable=other' 'the following components are not recognized: other'
-os::cmd::expect_failure_and_text 'openshift-sdn --config=${NODECONFIG} --disable=dns,proxy,plugins' 'at least one node component must be enabled \(dns, plugins, proxy\)'
+# TODO: We do not have node-configuration with the new local-up script
```
Comment (soltysh) [inline test/cmd/admin.sh]: Remove.

File: test/cmd/login.sh [hunk 16]
```diff
@@ -45,24 +45,24 @@ fi
 
 # remove self-provisioner role from user and test login prompt before creating any projects
 os::cmd::expect_success "oc adm policy remove-cluster-role-from-group self-provisioner system:authenticated:oauth --kubeconfig='${login_kubeconfig}'"
-os::cmd::expect_success_and_text "oc login --server=${KUBERNETES_MASTER} --certificate-authority='${MASTER_CONFIG_DIR}/ca.crt' -u no-project-test-user -p anything" "You don't have any projects. Contact your system administrator to request a project"
+os::cmd::expect_success_and_text "oc login --server=${KUBERNETES_MASTER} --certificate-authority='${MASTER_CONFIG_DIR}/server-ca.crt' -u no-project-test-user -p anything" "You don't have any projects. Contact your system administrator to request a project"
```
Comment (soltysh) [inline test/cmd/login.sh]: Why not creating a variable that you'll just use all over this script?

File: test/cmd/sdn.sh [hunk 17]
```diff
@@ -15,47 +15,47 @@ trap os::test::junit::reconcile_output EXIT
...
-os::test::junit::declare_suite_start "cmd/sdn/hostsubnets"
-# test-cmd environment has no nodes, hence no hostsubnets
-os::cmd::expect_success_and_text 'oc get hostsubnets' 'No resources found.'
-os::test::junit::declare_suite_end
-
-os::test::junit::declare_suite_start "cmd/sdn/egressnetworkpolicies"
-os::cmd::expect_success 'oc get egressnetworkpolicies'
-os::cmd::expect_success 'oc create -f test/integration/testdata/test-egress-network-policy.json'
-os::cmd::expect_success 'oc get egressnetworkpolicy default'
+#os::cmd::expect_success 'oc get clusternetworks'
```
Comment (soltysh) [inline test/cmd/sdn.sh]: Issue to get them back or remove if not needed, I don't see any clear information about it anywhere.
/cc @squeed

File: test/cmd/status.sh [hunk 18]
```diff
@@ -21,38 +21,38 @@ cp "${KUBECONFIG}" "${login_kubeconfig}"
 
 os::test::junit::declare_suite_start "cmd/status"
 # login and ensure no current projects exist
-os::cmd::expect_success "oc login --server=${KUBERNETES_MASTER} --certificate-authority='${MASTER_CONFIG_DIR}/server-ca.crt' -u test-user -p anything"
+os::cmd::expect_success_and_text "oc login --server=${KUBERNETES_MASTER} --certificate-authority='${MASTER_CONFIG_DIR}/server-ca.crt' -u test-user2 -p anything" 'You can try to create'
```
Comment (soltysh) [inline test/cmd/status.sh]: Why you're changing tests?

File: hack/local-up-master/openshift-apiserver-manifests/03_apiservice.yaml (hunk 9 above)
Comment (juanvallejo) [reply hack/local-up-master/openshift-apiserver-manifests/03_apiservice.yaml]: yeah, it is no longer supported in the latest version of hypershift

File: test/cmd/admin.sh (hunk 12 above)
Comment (juanvallejo) [reply test/cmd/admin.sh]: got the goahead from @deads2k via slack to update this to another identity. anypassword:... are apparently not available in the localup master

File: test/cmd/admin.sh (hunk 14 above)
Comment (juanvallejo) [reply test/cmd/admin.sh]: Same reason as above. For some reason I do not have anypassword: identities on localup master.

File: test/cmd/sdn.sh (hunk 17 above)
Comment (juanvallejo) [reply test/cmd/sdn.sh]: Ah I deleted this whole file. Linking Dan's comment: https://github.com/openshift/origin/pull/21206#discussion_r225298568

File: test/cmd/status.sh (hunk 18 above)
Comment (juanvallejo) [reply test/cmd/status.sh]: Debugging. I'll change it back. Need to figure out why the commands in this test behave differently when run via test-cmd.sh

File: hack/lib/cleanup.sh [hunk 19]
```diff
@@ -79,14 +79,10 @@ readonly -f os::cleanup::dump_etcd
...
 
-	local etcd_client_cert="${MASTER_CONFIG_DIR}/master.etcd-client.crt"
-	local etcd_client_key="${MASTER_CONFIG_DIR}/master.etcd-client.key"
-	local ca_bundle="${MASTER_CONFIG_DIR}/ca-bundle.crt"
-
 	os::util::ensure::built_binary_exists 'etcdhelper' >&2
 
-	etcdhelper --cert "${etcd_client_cert}" --key "${etcd_client_key}" \
-	           --cacert "${ca_bundle}" --endpoint "${full_url}" dump
+	etcdhelper --cert "${ETCD_CLIENT_CERT}" --key "${ETCD_CLIENT_KEY}" \
```
Comment (stevekuznetsov) [inline hack/lib/cleanup.sh]: Update the comment above if you change the API here.

What happens if these are unset?

File: hack/lib/cleanup.sh [hunk 20]
```diff
@@ -259,7 +255,6 @@ function os::cleanup::dump_events() {
 	if [[ -n "${ADMIN_KUBECONFIG:-}" ]]; then
 		kubeconfig="--config=${ADMIN_KUBECONFIG}"
 	fi
-	oc login -u system:admin ${kubeconfig:-}
```
Comment (stevekuznetsov) [inline hack/lib/cleanup.sh]: Why?

File: hack/lib/test/junit.sh [hunk 21]
```diff
@@ -158,7 +158,7 @@ function os::test::junit::generate_report() {
     if [[ -z "${JUNIT_REPORT_OUTPUT:-}" ||
           -n "${JUNIT_REPORT_OUTPUT:-}" && ! -s "${JUNIT_REPORT_OUTPUT:-}" ]]; then
         # we can't generate a report
-        return
+        return 0
```
Comment (stevekuznetsov) [inline hack/lib/test/junit.sh]: This is a no-op and should not be necessary

File: hack/lib/util/environment.sh [hunk 22]
```diff
@@ -253,6 +253,13 @@ function os::util::environment::setup_server_vars() {
     NODE_CONFIG_DIR="${SERVER_CONFIG_DIR}/node-${KUBELET_HOST}"
     export NODE_CONFIG_DIR
 
+    ETCD_CLIENT_CERT="${MASTER_CONFIG_DIR}/master.etcd-client.crt"
```
Comment (stevekuznetsov) [inline hack/lib/util/environment.sh]: Why?

File: hack/local-up-master/lib.sh [hunk 23]
```diff
@@ -20,23 +20,20 @@ MAX_TIME_FOR_URL_API_SERVER=${MAX_TIME_FOR_URL_API_SERVER:-1}
 
 #source "$(dirname "${BASH_SOURCE}")/../lib/init.sh"
 KUBE_ROOT=OS_ROOT
-source "${OS_ROOT}/hack/local-up-master/logging.sh"
-source "${OS_ROOT}/hack/local-up-master/util.sh"
-source "${OS_ROOT}/hack/local-up-master/etcd.sh"
+source "$(dirname "${BASH_SOURCE}")/logging.sh"
```
Comment (stevekuznetsov) [inline hack/local-up-master/lib.sh]: Why?

Comment (stevekuznetsov) [reply hack/local-up-master/lib.sh]: Also, `$BASH_SOURCE` is an array, so index it

File: hack/local-up-master/util.sh [hunk 24]
```diff
@@ -70,7 +70,9 @@ kube::util::trap_add() {
 
 # Opposite of kube::util::ensure-temp-dir()
 kube::util::cleanup-temp-dir() {
-  rm -rf "${KUBE_TEMP}"
+  if [[ ! -z ${KUBE_TEMP-} ]]; then
```
Comment (stevekuznetsov) [inline hack/local-up-master/util.sh]: `! -z` is `-n`

File: hack/local-up-master/util.sh [hunk 25]
```diff
@@ -70,7 +70,9 @@ kube::util::trap_add() {
...
+    rm -rf "${KUBE_TEMP}"
```
Comment (stevekuznetsov) [inline hack/local-up-master/util.sh]: Also this doesn't fail if there is no dir, so you don't technically need the check

File: hack/test-cmd.sh [hunk 26]
```diff
@@ -4,12 +4,15 @@
 # simple scenarios.  It does not require Docker so it can run in travis.
 source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 os::util::environment::setup_time_vars
+source "$(dirname "${BASH_SOURCE}")/local-up-master/lib.sh"
```
Comment (stevekuznetsov) [inline hack/test-cmd.sh]: One of the points of the `init.sh` is to give you `$OS_ROOT` ... why not use it?

File: hack/test-cmd.sh [hunk 27]
```diff
@@ -55,27 +58,45 @@ unset KUBECONFIG
 
 export ALLOWED_REGISTRIES='[{"domainName":"172.30.30.30:5000"},{"domainName":"myregistry.com"},{"domainName":"registry.centos.org"},{"domainName":"docker.io"},{"domainName":"gcr.io"},{"domainName":"quay.io"},{"domainName":"*.redhat.com"},{"domainName":"*.docker.io"},{"domainName":"registry.redhat.io"}]'
 
-os::start::configure_server
+eval $( ./hack/install-etcd.sh | tail -n1 )
```
Comment (stevekuznetsov) [inline hack/test-cmd.sh]: This immediately stands out as non-standard and potentially dangerous. Why?

File: hack/test-cmd.sh [hunk 28]
```diff
@@ -55,27 +58,45 @@ unset KUBECONFIG
...
+cp "${LOCALUP_CONFIG}/admin.kubeconfig" "${HOME}/.kube/non-default-config"
 export KUBECONFIG="${HOME}/.kube/non-default-config"
 export KUBERNETES_MASTER="${API_SCHEME}://${API_HOST}:${API_PORT}"
 
 # Store starting cluster RBAC to allow it to be restored during tests that mutate the global policy
 BASE_RBAC_DATA="$( mktemp "${BASETMPDIR}/base_rbac_data_XXXXX" )"
 export BASE_RBAC_DATA
+
 oc get clusterroles.rbac,clusterrolebindings.rbac -o yaml | grep -v resourceVersion > "${BASE_RBAC_DATA}"
+# these should probably be set up in localup::init_master
```
Comment (stevekuznetsov) [inline hack/test-cmd.sh]: Why do they have to be set at all?>

File: hack/test-cmd.sh [hunk 29]
```diff
@@ -89,7 +110,7 @@ for test in "${tests[@]}"; do
 
   os::test::junit::declare_suite_start "cmd/${namespace}-namespace-setup"
   # switch back to a standard identity. This prevents individual tests from changing contexts and messing up other tests
-  os::cmd::expect_success "oc login --server='${KUBERNETES_MASTER}' --certificate-authority='${MASTER_CONFIG_DIR}/ca.crt' -u test-user -p anything"
+  os::cmd::try_until_success "oc login --server=${KUBERNETES_MASTER} --certificate-authority ${MASTER_CONFIG_DIR}/server-ca.crt -u test-user -p anything" $(( 160 * second )) 0.25
```
Comment (stevekuznetsov) [inline hack/test-cmd.sh]: Why drop the quotes?

Comment (stevekuznetsov) [reply hack/test-cmd.sh]: Why the wait?

File: test/cmd/image-lookup.sh (hunk 2 above)
Comment (stevekuznetsov) [inline test/cmd/image-lookup.sh]: don't `exit` here, maybe just comment out

`exit` in non-top-level things could do weird stuff with the control flow IIRC

File: test/cmd/whoami.sh [hunk 30]
```diff
@@ -12,7 +12,7 @@ trap os::test::junit::reconcile_output EXIT
 
 os::test::junit::declare_suite_start "cmd/whoami"
 # This test validates the whoami command's --show-server flag
-os::cmd::expect_success_and_text 'oc whoami --show-server' 'http(s)?:\/\/[0-9\.]+\:[0-9]+'
+os::cmd::expect_success_and_text 'oc whoami --show-server' 'http(s)?:\/\/localhost\:[0-9]+'
```
Comment (stevekuznetsov) [inline test/cmd/whoami.sh]: Wait so this only ever runs locally now? Wasn't the point that you could run `test-cmd` against any cluster, too?

File: hack/test-cmd.sh (hunk 29 above)
Comment (juanvallejo) [reply hack/test-cmd.sh]: Changed it back. Had it waiting originally before adding preflight checks for the api server

File: hack/test-cmd.sh (hunk 28 above)
Comment (juanvallejo) [reply hack/test-cmd.sh]: Because I needed something that works first

File: test/cmd/sdn.sh (hunk 17 above)
Comment (juanvallejo) [reply test/cmd/sdn.sh]: @deads2k Is this test okay to remove / do we need to restore it back at some point?

Comment (deads2k) [reply test/cmd/sdn.sh]: delete the entire file and open a high severity bugzilla for networking team

Comment (deads2k) [reply test/cmd/sdn.sh]: @knobunc @squeed with the network APIs gone, we can no longer run these tests in test-cmd.sh.  @juanvallejo will open a bugzilla and reference the commit he has to delete them

File: test/cmd/whoami.sh (hunk 30 above)
Comment (deads2k) [reply test/cmd/whoami.sh]: > Wait so this only ever runs locally now? Wasn't the point that you could run `test-cmd` against any cluster, too?

We'll live with it to kill openshift start.  If someone wants it later, they can add it back.

File: test/cmd/sdn.sh (hunk 17 above)
Comment (juanvallejo) [reply test/cmd/sdn.sh]: Opened high severity bugzilla here: https://bugzilla.redhat.com/show_bug.cgi?id=1639835

File: hack/lib/test/junit.sh (hunk 21 above)
Comment (juanvallejo) [reply hack/lib/test/junit.sh]: Not returning 0 here was causing test-cmd to fail for me locally

**Comment by deads2k** [comment]
/lgtm

/hold 

Holding for @stevekuznetsov comments.  @juanvallejo once they are fixed up, feel free to unhold and re-apply labels

File: hack/lib/util/environment.sh (hunk 22 above)
Comment (juanvallejo) [reply hack/lib/util/environment.sh]: Because I can guarantee that MASTER_CONFIG_DIR will be set here. Also, these are needed for cleanup

File: hack/lib/cleanup.sh (hunk 19 above)
Comment (juanvallejo) [reply hack/lib/cleanup.sh]: Updated comment above to include new vars. If they are not set, this part of cleanup will fail. I believe these vars not being set was causing an extended_clusterup failure at some point

File: hack/test-cmd.sh (hunk 27 above)
Comment (juanvallejo) [reply hack/test-cmd.sh]: Updated this to explicitly re-export PATH based on the string returned by the `./hack/install-etcd.sh` script

**Comment by juanvallejo** [comment]
@soltysh @stevekuznetsov thanks for the review. Comments addressed.

File: hack/local-up-master/openshift-apiserver-manifests/03_apiservice.yaml (hunk 9 above)
Comment (soltysh) [reply hack/local-up-master/openshift-apiserver-manifests/03_apiservice.yaml]: Good to know, b/c I need to drop that from integration as well.

File: test/cmd/admin.sh (hunk 12 above)
Comment (soltysh) [reply test/cmd/admin.sh]: Hmmm... that's weird, do you have a followup for it?

File: test/cmd/policy.sh [hunk 31]
```diff
@@ -226,7 +226,7 @@ os::cmd::expect_success_and_text 'oc policy scc-subject-review -z default -g sys
 os::cmd::expect_failure_and_text 'oc policy scc-subject-review -u alice -z default -g system:authenticated -f ${OS_ROOT}/test/testdata/job.yaml' 'error: --user and --serviceaccount are mutually exclusive'
 os::cmd::expect_success_and_text 'oc policy scc-subject-review -z system:serviceaccount:alice:default -g system:authenticated -f ${OS_ROOT}/test/testdata/job.yaml' 'restricted'
 os::cmd::expect_success_and_text 'oc policy scc-subject-review -u alice -g system:authenticated -f ${OS_ROOT}/test/testdata/job.yaml' 'restricted'
-os::cmd::expect_failure_and_text 'oc policy scc-subject-review -u alice -g system:authenticated -n noexist -f ${OS_ROOT}/test/testdata/job.yaml' 'error: unable to compute Pod Security Policy Subject Review for "hello": namespaces "noexist" not found'
+os::cmd::expect_failure_and_text 'oc policy scc-subject-review -u alice -g system:authenticated -n noexist -f ${OS_ROOT}/test/testdata/job.yaml --loglevel 8' 'error: unable to compute Pod Security Policy Subject Review for "hello": namespaces "noexist" not found'
```
Comment (soltysh) [inline test/cmd/policy.sh]: Debug leftover still.

File: test/cmd/status.sh [hunk 32]
```diff
@@ -21,38 +21,38 @@ cp "${KUBECONFIG}" "${login_kubeconfig}"
...
 os::cmd::expect_success 'oc delete project --all'
 os::cmd::try_until_text "oc get projects -o jsonpath='{.items}'" "^\[\]$"
 os::cmd::expect_success 'oc logout'
 
 # remove self-provisioner role from user and test login prompt before creating any projects
 os::cmd::expect_success "oc adm policy remove-cluster-role-from-group self-provisioner system:authenticated:oauth --kubeconfig='${login_kubeconfig}'"
 
-# login as 'test-user'
-os::cmd::expect_success "oc login --server=${KUBERNETES_MASTER} --certificate-authority='${MASTER_CONFIG_DIR}/ca.crt' -u test-user -p anything"
+# login as 'test-user2'
```
Comment (soltysh) [inline test/cmd/status.sh]: Why this change?

File: test/cmd/explain.sh (hunk 3 above)
Comment (soltysh) [reply test/cmd/explain.sh]: Steve explicitly asked not to do `exit 0` since this may break stuff. Comment out what's not needed.

File: test/cmd/status.sh (hunk 32 above)
Comment (juanvallejo) [reply test/cmd/status.sh]: I wanted to use a different user from the one being used to log in in `test-cmd`. It was mostly a way for me to check that I was using the correct kubeconfig context in this test
